game/environment.go

```diff
@@ -277,12 +308,131 @@ func (g *environment) initVPNInterface(ipAddress string, port uint, vpnInterface
 	return nil
 }
 
-//TODO:WAITING FOR FRONTEND
-//func (g *environment) GetFrontend() *frontend.WebSite {
-//	return g.web
-//}
+func (gc *GameConfig) CreateVPNConfig(ctx context.Context, isRed bool, gameTag string, idUser string) (VPNConfig, error) {
+
+	var nicName string
+
+	var allowedIps string
+
+	var endpoint string
+	var hitNetworks string
+
+	if isRed {
+		nicName = fmt.Sprintf("%s_red", gameTag)
+
+		for key, _ := range gc.NetworksIP {
+			hitNetworks = gc.NetworksIP[key]
+			allowedIps = fmt.Sprintf("%s, %s", hitNetworks, gc.redVPNIp)
+			continue
+		}
+
+		allowedIps = gc.redVPNIp
+		endpoint = fmt.Sprintf("%s.defatt.haaukins.com:%d", gameTag, gc.redPort)
+	} else {
+
+		nicName = fmt.Sprintf("%s_blue", gameTag)
+		allowedIps = gc.blueVPNIp
+		endpoint = fmt.Sprintf("%s.defatt.haaukins.com:%d", gameTag, gc.bluePort)
+
+		//	10.20.30.
+	}
+
+	serverPubKey, err := gc.env.wg.GetPublicKey(ctx, &vpn.PubKeyReq{PubKeyName: nicName, PrivKeyName: nicName})
+	if err != nil {
+		log.Error().Err(err).Str("User", idUser).Msg("Err get public nicName wireguard")
+		return VPNConfig{}, err
+	}
+
+	_, err = gc.env.wg.GenPrivateKey(ctx, &vpn.PrivKeyReq{PrivateKeyName: gameTag + "_" + idUser + "_"})
+	if err != nil {
+		//fmt.Printf("Err gen private nicName wireguard  %v", err)
+		log.Error().Err(err).Str("User", idUser).Msg("Err gen private nicName wireguard")
+		return VPNConfig{}, err
+	}
+
+	//generate client public nicName
+	//log.Info().Msgf("Generating public nicName for team %s", evTag+"_"+team+"_"+strconv.Itoa(ipAddr))
+	_, err = gc.env.wg.GenPublicKey(ctx, &vpn.PubKeyReq{PubKeyName: gameTag + "_" + idUser + "_", PrivKeyName: gameTag + "_" + idUser + "_"})
```

Comment:
What does it return other than the error ? i think we can save a function call here